packages/memory-storage/src/resource-clients/request-queue.ts

```diff
@@ -365,6 +378,10 @@ export class RequestQueueClient extends BaseClient implements storage.RequestQue
             existingQueueById.handledRequestCount += 1;
         }
 
+        if (options.forefront) {
```

Comment:
If the request was already present in the queue, should `forefront` addRequest be just NOOP, or should it bump the request to the front of the queue?

Reply:
ideally it should bump it i guess

Reply:
It seems unspecified on platform, my gut feeling is that we should do whatever is simplest. 

Is it likely that somebody would intentionally do both a depth-first and breadth-first crawl?

Reply:
It feels quite normal to me that you discover a URL in one way (e.g. a random link on a homepage like in the footer or in some "last articles" section), then in a different code path/route handler, you see it again and consider that same URL as a priority and want to bump it. I would argue that "whatever is simplest" is a wrong measure, we want to do things that users would expect, not those that are simple for us.

If it's not trivial, we can double check with delivery what they consider as expected, I can imagine they will think the same as I did.

Reply:
I just checked the Apify API behavior, and the existing request doesn't get bumped with `forefront`, so `MemoryStorage` IMO shouldn't either.

Reply:
I wouldn't really expect a POST endpoint for insertion to behave like upsert, if that's what you mean. But I can imagine we can modify the forefront flag on an existing request.

Anyway, let's leave this for later and focus on problems that people actually reported.

Reply:
Ah, good point, there's the `forefront` option on `updateRequest` as well.